dipy/atlasing/bundles.py

```diff
@@ -0,0 +1,506 @@
+                            f" pair:{index+1}/{n_reg[i_step]}")
+
+                file = file_list[i] + '.trk'
+                bundle_obj = load_tractogram(file, reference='same',
+                                             bbox_valid_check=False)
+                static = bundle_obj.streamlines
+                header = create_tractogram_header(file,
+                                                  *bundle_obj.space_attributes)
+
+                file = file_list[j] + '.trk'
+                moving = load_tractogram(file, reference='same',
+                                         bbox_valid_check=False).streamlines
+
+                aligned, _, _, _ = slr_with_qbx(static, moving, x0='affine',
+                                                rm_small_clusters=1,
+                                                qbx_thr=[5])
+
+                points, offsets = unlist_streamlines(static)
+                static = relist_streamlines(points, offsets)
+                points, offsets = unlist_streamlines(aligned)
+                aligned = relist_streamlines(points, offsets)
+                points, offsets = unlist_streamlines(moving)
+                moving = relist_streamlines(points, offsets)
+
+                # Randomly skip steps if speciffied to get a sharper results
+                if skip_pairs and np.random.choice([True, False], 1)[0]:
+                    combined = combine_bundles(static, aligned, 'random_pick',
+                                               distance)
+                else:
+                    combined = combine_bundles(static, aligned, comb_method,
+                                               distance)
+
+                file = f'{step_dir}/bundle_{index+has_lonely}_prev_{i}_{j}'
+                new_file_list.append(file)
+                new_tractogram = StatefulTractogram(Streamlines(combined),
+                                                    reference=header,
+                                                    space=Space.RASMM)
+                save_trk(new_tractogram, file + ".trk", bbox_valid_check=False)
+                if save_temp and has_fury:
+                    show_bundles([static, moving, aligned], f'{file}_reg.png',
+                                 colors=[(0, 0, 1), (1, 0, 0), (0, 1, 0)])
+                    show_bundles([Streamlines(combined)], f'{file}.png')
+
+            file_list = new_file_list
+        save_trk(new_tractogram, f'{out_dir}/{bundle}.trk',
+                 bbox_valid_check=False)
+        atlas.append(Streamlines(combined))
+
+    if not save_temp:
+        rmtree(temp_dir)
+
+    if merge_out:
+        atlas_merged = np.concatenate(atlas)
+        file = f'{out_dir}/whole_brain.trk'
+        new_tractogram = StatefulTractogram(atlas_merged, reference=header,
+                                            space=Space.RASMM)
+        save_trk(new_tractogram, file, bbox_valid_check=False)
+        return atlas, atlas_merged
+    return atlas
```

Comment:
You can just `return atlas, atlas_merged`
`atlas_merged` can be None if there isn't one. This way the function output will be consistent (returns fixed number of objects).

Reply:
Will solve the issue in the workflow as well where you have written the same function call twice with if and else conditions.